RocketChat/Rocket.Chat
# [IMPROVE] Pinned Messages in E2E rooms

## Proposed changes (including videos or screenshots)



### Before
<img width="630" alt="Screenshot 2022-04-27 at 11 08 59 PM" src="https://user-images.githubusercontent.com/58601732/165591760-ab535329-05c9-4097-879c-1c4cdd57da1d.png">

### After
<img width="460" alt="Screenshot 2022-04-27 at 11 09 15 PM" src="https://user-images.githubusercontent.com/58601732/165591881-339694b8-f056-4c68-8704-2f86aafa11ab.png">

In Pinned Messages contextual bar -
<img width="414" alt="Screenshot 2022-04-27 at 11 29 25 PM" src="https://user-images.githubusercontent.com/58601732/165591936-610ebdff-41b3-4e6b-a80a-b2d32b4571f4.png">

If we don't have E2EE keys - 
<img width="831" alt="Screenshot 2022-04-27 at 11 08 44 PM" src="https://user-images.githubusercontent.com/58601732/165592021-3bf12df2-6b9e-4b3b-a48f-d88291c1f30e.png">




## Issue(s)


## Steps to test or reproduce


- Create an end to end encrypted room.
- Send message in the e2ee room and pin the message.
- Pinned message should follow the e2ee rules.

## Further comments
